File: CSharp/src/Runtime/Scripts/XTime/time.cs
// Copyright (c) 2026 FrameworkEase Technologies. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

using System;

namespace FrameworkEase.Universal.X.Utility
{
    public static partial class XTime
    {
        public static readonly DateTime Initial = new DateTime(1970, 1, 1) + TimeZoneInfo.Local.BaseUtcOffset;
        public static DateTime Current => DateTime.Now;
        public static int Seconds => (int)((DateTime.Now.Ticks - Initial.Ticks) / 10_000_000);
        public static long Milliseconds => (DateTime.Now.Ticks - Initial.Ticks) / 10_000;
        public static long Microseconds => (DateTime.Now.Ticks - Initial.Ticks) / 10;
    }
}

File: Go/XTime/time.go
// Copyright (c) 2026 FrameworkEase Technologies. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

package XTime

import (
	"time"
)

var initial = time.Date(1970, 1, 1, 0, 0, 0, 0, time.Local)

func Initial() time.Time { return initial }

func Current() time.Time { return time.Now() }

func Seconds() int {
	now := time.Now()
	return int(now.Sub(Initial()).Seconds())
}

func Milliseconds() int {
	now := time.Now()
	return int(now.Sub(Initial()).Milliseconds())
}

func Microseconds() int {
	now := time.Now()
	return int(now.Sub(Initial()).Microseconds())
}
